pkg/gardenlet/controller/shoot/care/reconciler.go

```diff
@@ -85,9 +95,17 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 
 	// if shoot is no longer managed by this gardenlet (e.g., due to migration to another seed) then don't requeue.
 	if ptr.Deref(shoot.Spec.SeedName, "") != r.SeedName {
+		if shoot.Status.TechnicalID != "" {
+			// cleanup mapping
+			r.namespaceToShootName.Delete(shoot.Status.TechnicalID)
+		}
```

Comment:
Why is this part of this other `if` condition? What is the relation to the `.spec.seedName`?